catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/UpdateOperations.java

```diff
@@ -460,34 +518,40 @@ private UpdateRequest populateUpdateRequestPolicyMap(UpdateRequest updateRequest
         return updateRequest;
     }
 
-    private Map<String, Metacard> populateMetacards(UpdateRequest updateRequest)
-            throws IngestException {
-        if (frameworkProperties.getPolicyPlugins()
-                .isEmpty()) {
-            return new HashMap<>();
-        }
+    private UpdateRequest populateMetacards(UpdateRequest updateRequest) throws IngestException {
+        final String attributeName = updateRequest.getAttributeName();
 
         QueryRequestImpl queryRequest = createQueryRequest(updateRequest);
-
+        QueryResponse query;
         try {
-            QueryResponse query = queryOperations.doQuery(queryRequest,
+            query = queryOperations.doQuery(queryRequest,
                     frameworkProperties.getFederationStrategy(),
                     false);
-            List<Result> results = query.getResults();
-            if (results.size() != updateRequest.getUpdates()
-                    .size()) {
-                throw new IngestException("Unable to find all metacards in update request.");
-            }
-
-            return results.stream()
-                    .map(Result::getMetacard)
-                    .collect(Collectors.toMap(metacard -> getAttributeStringValue(metacard,
-                            updateRequest.getAttributeName()), m -> m));
         } catch (FederationException e) {
             LOGGER.warn("Unable to complete query for updated metacards.", e);
+            throw new IngestException("Exception during runtime while performing update");
         }
 
-        return new HashMap<>();
+        List<Result> results = query.getResults();
+        if (notFoundValidMetacards(updateRequest, query)) {
+            logFailedQueryInfo(updateRequest, query);
+            throw new IngestException("Unable to find all metacards in update request.");
+        }
+
+        updateRequest = rewriteRequestToAvoidHistoryConflicts(updateRequest, query);
+
+        HashMap<String, Metacard> metacardMap = new HashMap<>(results.stream()
+                .map(Result::getMetacard)
+                .collect(Collectors.toMap(metacard -> getAttributeStringValue(metacard,
```

Comment:
`toMap` will throw an exception if the request accidentally includes two metacards with the same value for the update attribute.  Are we verifying that before we get here?  It does not look like it.